api4/emoji.go

```diff
@@ -28,6 +30,11 @@ func (api *API) InitEmoji() {
 }
 
 func createEmoji(c *Context, w http.ResponseWriter, r *http.Request) {
+	defer func() {
+		io.Copy(ioutil.Discard, r.Body)
```

Comment:
Do we need to check if `r.Body` is `nil` here? Also, any reason not to just `defer consumeAndClose()`?

https://github.com/mattermost/mattermost-server/blob/e8b8ef394b0bfb9a0798f43e282804c6b29390f8/app/server.go#L264-L269

Reply:
No for the server part.

From the http.Request source code:
```
        // For server requests the Request Body is always non-nil
        // but will return EOF immediately when no body is present.
        // The Server will close the request body. The ServeHTTP
        // Handler does not need to.
```

Reply:
Actually I don't need the Close either.